test/jdk/java/io/SequenceInputStream/TransferTo.java

```diff
@@ -126,6 +128,49 @@ public void testStreamContents() throws Exception {
                 outputStreamProvider, createRandomBytes(4096, 0), 0, 4096);
     }
 
+    /*
+     * Special case: Assert subsequent input stream is read when preceding stream already was MAX_VALUE long.
+     * Note: Not testing actual content as it requires multiple GBs of memory and long time.
+     */
+    @Test
+    public void testHugeStream() throws Exception {
+        InputStream is1 = repeat(0, Long.MAX_VALUE);
+        InputStream is2 = repeat(0, 1);
+        assertNotEquals(is1.available(), 0);
```

Comment:
Wdyt of asserting the expected value with assertEquals or removing the assertions before .transferTo altogether?
```suggestion
        assertEquals(is1.available(), Integer.MAX_VALUE, "The stream should have Long.MAX_VALUE bytes .available() before .transferTo (.available returns int)");
```

Reply:
The reader shall understand what the test is good for and what it works like. There is no benefit in checking the *actual* value (there is no need to explicitly fail if `available` returns something *else*, as long as it is *not zero*), so this would confuse the reader more than it helps. I kept the before-assertations solely to proof that the before-value of `available()` is *not the default value* of `InputStream`, hence to proof that the source code of this test is actually able to detect a *change*. We can remove this, it is not needed, but it makes it easier *to understand* what this test actually expects *to change*.